src/utils/general-utils.ts

```diff
@@ -41,11 +41,16 @@ export function mustExist<Type>(
 export const getNetworkFromSignerOrProvider = async (
   signerOrProvider: ethers.providers.Provider | ethers.Signer
 ): Promise<Network> => {
-  if (signerOrProvider instanceof ethers.providers.Provider) {
-    const chainId = (await signerOrProvider.getNetwork()).chainId;
+  // providers have a property _isProvider, which is more reliable than instanceOf
+  if ((signerOrProvider as ethers.providers.Provider)._isProvider) {
```

Comment:
Please use `Provider.isProvider` or `Signer.isSigner` instead
```suggestion
  import {Provider} from "@ethersproject/providers";
  // ...
  if (Provider.isProvider(signerOrProvider)) {
```
I think it would be better to use a ternary here to get the provider
```suggestion
  const provider: Provider = Provider.isProvider(signerOrProvider)
    ? signerOrProvider
    : signerOrProvider.provider;
  const chainId = (await provider.getNetwork()).chainId;
  return CHAIN_ID_TO_NETWORK_NAME[chainId];
```